app/api/src/endpoints/v1/static_layers.py

```diff
@@ -1,26 +1,101 @@
-from typing import Any
-
-from fastapi import APIRouter, Depends
+import geopandas
+from fastapi import APIRouter, Depends, HTTPException, UploadFile
 from sqlalchemy.ext.asyncio import AsyncSession
 
 from src import crud
-from src.crud.crud_customization import customization, dynamic_customization
 from src.db import models
-
-from src.endpoints import deps
 from src.db.models.config_validation import *
-from fastapi.encoders import jsonable_encoder
-from src.utils import return_geojson_or_geobuf
+from src.db.session import legacy_engine
+from src.endpoints import deps
+from src.utils import generate_static_layer_table_name, return_geojson_or_geobuf
 
 router = APIRouter()
-    
-@router.get("/static/{layer_name}") 
-async def static_vector_layer(*, 
+
+
+@router.get("/static/{layer_name}")
+async def static_vector_layer(
+    *,
     db: AsyncSession = Depends(deps.get_db),
-    current_user: models.User =  Depends(deps.get_current_active_user),
-    layer_name: str, 
-    return_type: str
+    current_user: models.User = Depends(deps.get_current_active_user),
+    layer_name: str,
+    return_type: str,
 ):
     """Return selected layers in different vector formats"""
-    layer_to_return = await crud.layer.static_vector_layer(db, current_user, layer_name, return_type)
-    return return_geojson_or_geobuf(layer_to_return, return_type)
\ No newline at end of file
+    layer_to_return = await crud.layer.static_vector_layer(
+        db, current_user, layer_name, return_type
+    )
+    return return_geojson_or_geobuf(layer_to_return, return_type)
+
+
+router2 = APIRouter()
+
+
+@router2.post("/static")
+async def uplaod_static_layer(
```

Comment:
### Auto Deleting UploadFile
Seems the UploadFile will be deleted after closing the session automatically as it is a SpooledTemporaryFile [ref](https://fastapi.tiangolo.com/tutorial/request-files/#uploadfile). I'm not sure about when the file exceeds it's max_size. Because then it will be written to disk. [Someone says](https://stackoverflow.com/a/59393654/1951027) the file will be deleted even it was written to disk. I didn't see any reference inside the [docs](https://docs.python.org/3/library/tempfile.html#tempfile.SpooledTemporaryFile). Maybe we should test or check more docs.

Reply:
This is great!

Reply:
### Opening different file types inside zip file
The GeoPandas now can open different Geo data files inside zip file.
But the problem about zip files is that if the files are under a directory inside zip file the GeoPandas doesn't open them automatically, even if there is only one directory. It can determine the directory if we specify the directory name for it.
As for example in the sample.zip, It couldn't open it. In such situations, if we automatically detect the directory name and specify it for read_file() then it would open it.
I'm going to check the [documentation](https://geopandas.org/en/stable/docs/user_guide/io.html) for better solution.

Reply:
Ok, great thanks! An alternative would be reading the Zipfile with another library and then unpacking it. Then it could be read with Geopandas, I guess.

Reply:
The only way I think is to:
1. detect if the zip file contains only one directory
2. save it to disk
3. open it by geopandas.read_file() by giving the sub directory.
4. remove the file.

Reply:
It should be like this I think:
![Open Uploaded PostGIS File](https://user-images.githubusercontent.com/3726265/181187410-84ab0612-addd-4a7c-9f01-112d33c82ca6.png)

Reply:
Thanks for this great flow chart. I believe this is exactly how we should do it.

Reply:
:heavy_check_mark: Done
If the file type is not zip, it won't try to save file to disk. It uses the default functionality of SpooledTemporaryFile as it will be destroyed automatically.

Reply:
Makes sense!